tests/e2e.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+ROOT="$(git rev-parse --show-toplevel)"
+
+# Apply the GCS emulator configuration
+kubectl apply -f "${ROOT}/tests/gcs-emulator.yaml"
+
+# Wait for the deployment to be ready
+echo "Waiting for GCS emulator deployment to be ready..."
+kubectl wait --for=condition=available --timeout=300s deployment/gcs-emulator
+
+# Check the deployment status
+if [ $? -eq 0 ]; then
+    echo "GCS emulator deployment is ready"
+else
+    echo "Error: GCS emulator deployment failed to become ready within the timeout period"
+    exit 1
+fi
+
+ko apply -R -f ${ROOT}/dev/step-action/
+
+ko create -f ${ROOT}/tests/test-pipeline-gcs.yaml
+
+# Check the pipelinerun status
+echo "Waiting for pipelinerun to complete..."
+kubectl wait --for=condition=succeeded --timeout=600s pipelinerun --all
+
+if [ $? -eq 0 ]; then
+    echo "Pipelinerun completed successfully"
+    exit 0
+else
+    echo "Error: Pipelinerun failed or timed out"
+    kubectl get pipelinerun
+    kubectl describe pipelinerun
+    kubectl get taskrun
+    kubectl describe taskrun
+    exit 1
+fi
+
```

Comment:
I didn't get what are we asserting here ?
If I'm not wrong we are just creating GCS emulateor deployment and installing fetch and upload stepActions and finally creating pipelineruns

Reply:
We are checking whether pipelinerun succeeded or not.

Reply:
And also in the last step, we are verifying whether cache was uploaded and fetched or not. It has different workspace where we fetch the cache.